projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -1281,14 +1284,15 @@ private static V1EnvVar envVar(Map.Entry<String, String> entry) {
     }
 
     /**
-     * Removes secrete with given name if it exists.
+     * Removes secret with given name if it exists.
      */
     public void removeSecretData(String name) throws ApiException {
         log.debug("Deleting k8s secret: {}", name);
         try {
             var status = new CoreV1Api(client).deleteNamespacedSecret(name, this.namespace, null, null, null, null, null, null);
             log.debug("Deleted k8s secret: {}, status: {}", name, status);
         } catch (ApiException e) {
+            log.error("Error while trying to remove K8S secret", e);
```

Comment:
That should not be error. If status is 404 (Not Found), this si an expected case. But we will print an error necessarily. 

So I think this should be in the else statement.

Reply:
Changed.